akka-sample-distributed-data-scala/src/main/scala/sample/distributeddata/ReplicatedCache.scala

```diff
@@ -1,52 +1,66 @@
 package sample.distributeddata
 
-import akka.actor.Actor
-import akka.actor.ActorRef
-import akka.actor.Props
-import akka.cluster.Cluster
-import akka.cluster.ddata.DistributedData
+import akka.actor.typed.ActorRef
+import akka.actor.typed.Behavior
+import akka.actor.typed.scaladsl.Behaviors
 import akka.cluster.ddata.LWWMap
 import akka.cluster.ddata.LWWMapKey
+import akka.cluster.ddata.Replicator._
+import akka.cluster.ddata.SelfUniqueAddress
+import akka.cluster.ddata.typed.scaladsl.DistributedData
+import akka.cluster.ddata.typed.scaladsl.Replicator.{ Update, Get }
 
 object ReplicatedCache {
+  sealed trait Command
+  final case class PutInCache(key: String, value: Any) extends Command
+  final case class GetFromCache(key: String, replyTo: ActorRef[Cached]) extends Command
+  final case class Cached(key: String, value: Option[Any])
+  final case class Evict(key: String) extends Command
+  private sealed trait InternalCommand extends Command
+  private case class InternalGetResponse(key: String, replyTo: ActorRef[Cached], rsp: GetResponse[LWWMap[String, Any]])
+      extends InternalCommand
+  private case class InternalUpdateResponse(rsp: UpdateResponse[LWWMap[String, Any]]) extends InternalCommand
```

Comment:
Tricky (nitpick) naming aspect: these are phrased as events but the supertype is `InternalCommand`. Changing`InternalCommand` to `InternalEvent` is weird though because it is a subtype of `Command`.  One way to get around it we used in the training materials is to name these as commands instead of events - for example `ConfirmUpdate`, `ForwardToClient`.

Reply:
I based this on the example that's front and centre of the type ddata docs: https://doc.akka.io/docs/akka/2.6/typed/distributed-data.html#using-the-replicator.

Should we change both?

Reply:
Nah, let's leave it be it's just a minor detail.